activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -230,7 +230,15 @@ def column_for_attribute(name)
     #   person.respond_to(:nothing) # => false
     def respond_to?(name, include_private = false)
       return false unless super
-      name = name.to_s
+
+      case name
```

Comment:
We're calling respond_to? against an argument passed in. This is to see if the object is a model. A normal ruby Object needs a symbol key for that method. Changing to a frozen string would be a breaking change.